examples/token-classification/utils_ner.py

```diff
@@ -358,6 +358,8 @@ def gen():
                     ),
                 )
 
+            self.dataset = self.dataset.map(lambda features, labels: self.count_tokens_in_example((features, labels)))
```

Comment:
Adding the info as a feature is not a valuable solution. Features are not always dictionaries and might be a list, tuple or simply positional arguments.

Reply:
OK, I thought that it was fixed to be a dictionary to simplify the work on example scripts. Currently, I don't have a better idea about other options.

Reply:
Do you think sth like:

    {
        'batch': the unmodified batch 
        'nb_instances_in_batch': the value we dynamically compute for batch
    }

is acceptable? This way, we don't rely on the structure of the datasets/batch.

Reply:
Better! but the problem is that we have to re-adapt the training to take this new datatype, which will bring a bit too much of changes in the way the trainer uses the data and has to be repercuted to all the other tasks. A bit too breaking change.

The best solution would be to compute this size on the fly just before the scaled loss computation. I mean, we do it in the loss function why not directly here. It is just matter of computing the number of token that have a label != 100.

Reply:
I think with some checks,  like if the type of a batch is `dict` and if `nb_instances_in_example` is in that `dict`. If so, we can use it to compute the information. Otherwise, `nb_instances_in_example` is assumed to be `1`, and we just compute the number of examples. This won't affect other tasks' dataset format, and the change in tf trainer is still minimal.

I can try to have a version with this format and see if it is acceptable. OK for you?

Reply:
Sounds a much better solution!

We should see afterward if we can put this computation outside of the utils_ner because not everybody uses this file for training a token classifier and if they don't they will get the bug of loss computation.